### PR TITLE
PR: Migrate Help plugin to new API

### LINKED ISSUE #12186: Move Help plugin to use new API
- [ ] Move to new API
- [ ] Make sure tests pass

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #12186 
Depends on #11741

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

**Comment by goanpeca** [comment]
Rebased and ready for review @ccordoba12

**Comment by goanpeca** [comment]
@dalthviz this one is also ready for a check. Probably the dock issue will be present here, so no need to report it :-) 

--- 

@ccordoba12 this PR also has all newly updated docstrings. I left some `??`in things I was not sure, so any suggestions there are appreciated.
---

Thanks

File: spyder/plugins/help/plugin.py [hunk 1]
```diff
@@ -4,734 +4,263 @@
...
-
-    @Slot()
-    def toggle_locked(self):
+    def show_tutorial(self):
+        """Show the spyder tutorial on the help browser."""
+        self.switch_to_plugin()
+        self.get_widget().show_tutorial()
+
+    def show_intro_message(self):
+        """Show the introduction message on the help browser."""
```
[the remark this answers is not in the document]
Comment (ccordoba12) [reply spyder/plugins/help/plugin.py:176]: Yes, it comes from here

Selección_003

And now that I took a deeper look at this, the right docstring needs to be:

     """Show the IPython introduction message."""

File: spyder/plugins/help/widgets.py [hunk 2]
```diff
@@ -179,3 +236,819 @@ def set_text(self, text, is_code):
...
+    def set_rich_text_html(self, html_text, base_url):
+        """
+        Set rich text.
+
+        Parameters
+        ----------
+        html_text: str
+            Html string.
+        base_url: str
+            ??
```
[the remark this answers is not in the document]
Comment (ccordoba12) [reply spyder/plugins/help/widgets.py:646]: According to Qt docs:

    Location of stylesheets and images to load in the page.

File: spyder/plugins/help/plugin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (goanpeca) [reply spyder/plugins/help/plugin.py:176]: Fixed

File: spyder/plugins/help/widgets.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (goanpeca) [reply spyder/plugins/help/widgets.py:646]: Thanks! fixed

**Comment by goanpeca** [comment]
>Almost done! After these minor changes, please do the respective squash, so I can merge this one.

Rebased @ccordoba12

**Pull request merged**